Client/core/CLocalization.cpp

```diff
@@ -41,10 +41,7 @@ CLocalization::CLocalization(const SString& strLocale, const SString& strLocaleP
 
 CLocalization::~CLocalization()
 {
-    for (auto iter : m_LanguageMap)
```

Comment:
`m_pCurrentLang` became dangling (and a potential UAF).. you didn't account for it in the destructor after converting to unique_ptr. If accessed between destructing `m_LanguageMap` and `m_DictManager`, you will also have an UAF.

Reply:
> became dangling, or was already dangling? I didn't touch m_pCurrentLang

@Lpsd

The old code manually deleted all CLanguage* objects.. `m_pCurrentLang` pointed to one of those objects. With the old manual deletion it was already dangling, but that didnt really matter because the destrctor was about to finish anyways.

You went from manual deletion to unique_ptr. When m_LanguageMap destructs, it destroys all of the unique ptr objects, but m_pCurrentLang still points to one of them

Between line 42 (destructor starts) and when m_LanguageMap destructs (line 44), m_pCurrentLang points to valid memory. After m_LanguageMap destructs, m_pCurrentLang becomes dangling

If anything touches m_pCurrentLang during shutdown, there you got it.